doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+# HTTP Transport Adapter
+The Azure SDK for C++ creates and sends HTTP request using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc). It defines the interface that the Azure SDK for C++ will call to put the HTTP request into the wire.
+
+## Default HTTP Transport Adapter
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl transport adapter
+- WinHttp transport adapter
+
+Azure SDK for C++ users can create and use their own transport adapter as well [see below](#Building-a-custom-HTTP-Transport-Adapter).
+
+For the simplest case, when there was not any specific configuration set for the Azure SDK for C++, the default transport adapter will be selected based on the OS. The WinHTTP transport adapter will be used for Windows-based systems (making libcurl an optional dependency). The libcurl transport adapter will be selected for any non-windows-based system.
+
+## Building HTTP transport adapter
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
```

Comment:
Same here and elsewhere. Add new lines between title and body.
```suggestion
## Building HTTP transport adapter

The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
```

Reply:
Curious. Why ?

Reply:
Markdown linter rule `MD022`:
https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md022---headings-should-be-surrounded-by-blank-lines